onnxruntime/contrib_ops/cuda/bert/attention.cc

```diff
@@ -70,6 +95,27 @@ Status Attention<T>::ComputeInternal(OpKernelContext* context) const {
   int past_sequence_length = 0;
   Tensor* present = GetPresent(context, past, batch_size, head_size, sequence_length, past_sequence_length);
 
+  // Check whether we can use fused kernel
+  int sm = device_prop.major * 10 + device_prop.minor;
+  bool use_fused_runner = (!disable_fused_runner_ &&
+                           nullptr != mask_index && mask_index->Shape().NumDimensions() == 1 &&
+                           nullptr == past &&
+                           nullptr == present &&
+                           nullptr == extra_add_qk &&
+                           !is_unidirectional_ &&
+                           HasFusedFp16Kernel(sm, head_size, sequence_length));
+
+  MHARunner* fused_runner = nullptr;
+  if (use_fused_runner) {
+    if (nullptr == fused_fp16_runner_.get()) {
+      fused_fp16_runner_.reset(new FusedMHARunnerFP16v2(num_heads_, head_size, sm));
```

Comment:
Can `head_size` vary across Run() calls ? My understanding is it can if `hidden_size` can be allowed to change (when `hidden_size` is allowed to be a dynamic dim value). If so, `fused_fp16_runner_` has to be initialized per-compute call right ? It seems dangerous to allow it to be a mutable class member.

Reply:
head_size right now is fixed: only 64 is supported by fused kernel. In the future, if we support multiple head_size like 32 & 64, we shall consider your suggestion.